device/tt_silicon_driver.cpp

```diff
@@ -1052,20 +949,22 @@ void tt_SiliconDevice::write_buffer(
     std::uint16_t channel,
     chip_id_t src_device_id) {
 
-    void * user_scratchspace = nullptr;
-    if(hugepage_mapping.at(src_device_id).at(channel)) {
-      log_assert(size <= HUGEPAGE_REGION_SIZE, "write_buffer data has larger size {} than destination buffer {}", size, HUGEPAGE_REGION_SIZE);
-      log_debug(LogSiliconDriver, "Using hugepage mapping at address {} offset {} chan {} size {}",
-        hugepage_mapping.at(src_device_id).at(channel),
-        (address & HUGEPAGE_MAP_MASK),
+    log_assert(m_pci_device_map.find(src_device_id) != m_pci_device_map.end(), "write_buffer: Device id is not a MMIO device");
+
+    hugepage_mapping hugepage_map = m_pci_device_map.at(src_device_id)->get_hugepage_mapping(channel);
+    log_assert(hugepage_map.mapping, "write_buffer: Hugepages are not allocated for src_device_id: {} ch: {}."
+                                             " - Ensure sufficient number of Hugepages installed per device (1 per host mem ch, per device)",
```

Comment:
A lot of this log_* lines are not aligned properly when it breaks into multiple lines, but if that is how linter has formatted this then don't change it

Reply:
Linter is still not turned on for these files.
I won't fix this manually, we should turn on the linter gradually.